tutorial/input_output.py

```diff
@@ -0,0 +1,234 @@
+                "Relative to the current working directory, which is the location of the current Python script being run.": "Correct!",
+                "Relative to the user's home directory": "Wrong!",
+            },
+            correct_answer="Relative to the current working directory, which is the location of the current Python script being run.",
+            shuffle=True,
+        )
+
+        super().__init__(questions=[q1, q2, q3])
+
+
+class ReadFiles(Quiz):
+    def __init__(self, title=""):
+        q1 = Question(
+            question="Can you read from a file before calling <code>open</code>?",
+            options={
+                "Yes": "Wrong, if the file is not open, we cannot access its contents.",
+                "No": "Correct, we need to open the file first.",
+            },
+            correct_answer="No",
+            shuffle=True,
+        )
+
+        q2 = Question(
+            question="The function <code>readlines</code> reads the entire content of a text file into <b>one</string>. Is this correct?",
+            options={
+                "No": "Correct! It reads the file line by line.",
+                "Yes": "Wrong! It reads the file line by line and returns a list of <code>str</code>, with one element for each line.",
+            },
+            correct_answer="No",
+            shuffle=True,
+        )
+
+        super().__init__(questions=[q1, q2])
+
+
+class WriteFiles(Quiz):
+    def __init__(self, title=""):
+        q1 = Question(
+            question="""What does <code>w</code> in the second argument of <code>open</code> do:  <code>open(path, "w")</code>?""",
+            options={
+                "It opens the file for writing.": "Correct.",
+                "It writes a w next to each line of the file": "Wrong, it opens the file for reading.",
+            },
+            correct_answer="It opens the file for writing.",
+            shuffle=True,
+        )
+
+        q2 = Question(
+            question="What function do we use on a file object to write a list of strings line-by-line",
+            options={
+                "<code>write</code>": "Wrong, this function only writes a single string.",
+                "<code>writestrings</code>": "Wrong, this function does not exist.",
+                "<code>writelines</code>": "Correct.",
+            },
+            correct_answer="<code>writelines</code>",
+            shuffle=True,
+        )
+
+        super().__init__(questions=[q1, q2])
+
+
+class ContextManagers(Quiz):
+    def __init__(self, title=""):
+        q1 = Question(
+            question="Do you need to call <code>close</code> on a file object when using a context manager?",
+            options={
+                "Yes": "Wrong! The context manager will handle the closing of a file when the context manager scope ends.",
+                "No": "Correct! The context manager automatically calls close when leaving the scope.",
+            },
+            correct_answer="No.",
```

Comment:
Remove the `.`, otherwise the answer doesn't match.

Reply:
Fixed in 4fb2982